Add JSON report (#889)
MarketSquare/robotframework-robocop: docs/releasenotes/4.0.0.rst

Other features
==============

New report ``json_report`` (#663)
---------------------------------

> Added new ``json_report`` report that produces file with issues in JSON format. It's non default report which needs
> to be enabled using report name (it will not be loaded with ``--reports all``)::

Reviewer: ```suggestion
Added new ``json_report`` report that produces file with issues in JSON format. It's a non-default report which needs
to be called explicitly by name (it will not be loaded with ``--reports all``)::
```

    robocop --reports json_report .

> It will save file to current working directory under ``robocop_output.json`` filename. Output directory and filename
> can be configured::

Reviewer: ```suggestion
It will save the results to a JSON file in the current working directory under ``robocop_output.json`` filename.
Output directory and filename can be configured with::
```

    robocop --configure json_report:output_dir:output --configure json_report:report_filename:issues.json --reports json_report .

> 

Reviewer: Shouldn't we mention here that the previous `json_report` has now a new name? So that the users know they need to update it.

Author: Previous one was internal report and shouldn't be used by users anyway (at least I hope..). But I will mention this in our release docs
